vocab.py

```diff
@@ -20,9 +28,69 @@ class ValidDataSourceName(Enum):
     Each data source name should be defined in ValidDataSourceName.
     NAME_OF_DATA_SOURCE = auto()
     """
-    
+
     CANVAS_API = auto()
     CANVAS_ZOOM_MEETINGS = auto()
     KALTURA_API = auto()
     UNIZIN_DATA_PLATFORM_EVENTS = auto()
     UNIZIN_DATA_WAREHOUSE = auto()
+
+
+class DataSourceStatus:
+    def __init__(
+            self,
+            dataSource: ValidDataSourceName = NotImplemented,
+            timestamp: datetime = NotImplemented
+    ) -> None:
+        self.dataSource: ValidDataSourceName
+        self.timestamp: datetime
+        self.setDataSource(dataSource)
+        self.setTimestamp(timestamp)
+
+    def setDataSource(
+            self,
+            dataSource: ValidDataSourceName = NotImplemented
+    ) -> DataSourceStatus:
+        if (dataSource is NotImplemented):
+            raise ValueError('dataSource must be specified')
+        self.dataSource = dataSource
+        return self
+
+    def getDataSource(self) -> ValidDataSourceName:
+        return self.dataSource
+
+    def setTimestamp(self, timestamp: datetime = NotImplemented) -> DataSourceStatus:
+        if (timestamp is NotImplemented):
+            self.timestamp = pd.to_datetime(time.time(), unit='s', utc=True)
+        elif (timestamp.tzinfo is not UTC):
+            raise ValueError('timestamp must have UTC time zone')
+        else:
+            self.timestamp = timestamp
+        return self
+
+    def getTimestamp(self) -> datetime:
+        return self.timestamp
+
+    def copy(self) -> Dict:
+        '''
+        For backwards compatibility with code that expects to copy this object as a dictionary.
+
+        TODO: Update other code to not need this, then remove this method.
+
+        :return: Dict of object properties
+        '''
+        return {
+            'data_source_name': self.dataSource,
+            'data_updated_at': self.timestamp
+        }
```

Comment:
@lsloan, I know you're still working on updating this, but I was thinking about it a little bit. You may still need something like this for it to be compatible with the use of `pd.DataFrame` in `run_jobs.py`. Even if not, it still be useful to have a `to_dict` or similar method for testing and logging purposes.

Reply:
Yes, you're right.  We're thinking along the same lines.  I ended up keeping `copy()`, with a slight change.